Acromag channel handling

cristinasewell opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
## Description
Refactor the way we create entries from the questionnaire a bit.
Use the `AcromagChannelInput` and `AcromagChannelOutput` for `device_class` in the translations when creating a db item from questionnaire.
Handle the `BeckhoffAxis` motor entry creation from the questionnaire.
This PR will work with https://github.com/pcdshub/pcdsdevices/pull/664 
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Trying to tackle https://github.com/pcdshub/Bug-Reports-and-Requests/issues/51
Trying to tackle https://github.com/pcdshub/Bug-Reports-and-Requests/issues/53
## How Has This Been Tested?
```
In [2]: import hutch_python.qs_load as qs_load

In [3]: experiment = 'xpplv6818'

In [4]: obj = qs_load.get_qs_objs(experiment)

In [6]: a = obj.get('ao1_6')
In [7]: a
Out[7]:

ao1_6
-----
channel: 0.0 [Volts]

In [8]: a.get()
Out[8]: AcromagChannelOutputTuple(channel=0.0)
```
This test has been tried before in a different PR that had the same implementation for handling the `Bekhoff` axis:
```
In [1]: import hutch_python.qs_load as qs_load

In [2]: experiment = 'xpplv6818'

In [3]: obj = qs_load.get_qs_objs(experiment)

In [4]: v = obj.get('vh_y_donotuse')

In [5]: v.check_value
Out[5]: <bound method EpicsMotorInterface.check_value of BeckhoffAxis(HXX:VON_HAMOS:MMS:01, name=vh_y_donotuse)>

In [6]: vx = obj.get('vh_x')

In [7]: vx.check_value
Out[7]: <bound method PCDSMotorBase.check_value of IMS(XPP:USR:MMS:17, name=vh_x)>
```
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Where Has This Been Documented?
<!--  Include where the changes made have been documented. -->
<!--  This can simply be  a comment in the code or updating a docstring -->
Docstring
<!--
## Screenshots (if appropriate):
-->

cristinasewell commented on happi/backends/qs_db.py:
```diff
@@ -244,16 +244,29 @@ def _create_db_item(info: dict,
 
+        pv_prefix = info['pvbase']
+
+        if class_name and class_name in (
```
Maybe I should be refactoring things a bit here to handle the code for creating entries separately for every device or something of this sort?
It could get messy here very soon the way I am doing it now...
I am also working on getting the `mpod` here as well, and I'll have to handle some information specific for `mpod`  too, as it has instead of `pvbase` -> `pvname` and I also have to send custom `kwargs` ..

Any suggestion here would be appreciated :)

cristinasewell commented:
Ok I might have made it worse in f9cae57 than it was before..... not sure what to think of it, I'm duplicating some code this way a little bit and don't like it.  Any feedback/suggestions @klauer , @ZLLentz , @ZryletTC 

Still need to add `mpod` stuff here but wanted to see if I should proceed with this implementation or not first

klauer commented:
A minor suggestion, but I think you can drastically reduce the "translation" dictionary I made to something much more simple:
```python
DEFAULT_TRANSLATIONS = {
    'motors': _create_motor_callable,
    ...
}
```

Where it's up to each of the callables to take in the info dictionary and return a happi JSON backend-compatible dictionary. The callable gets to choose the container, class name, device name, etc.

cristinasewell commented:
> Overall, the idea and implementation are pretty good 👍
> 
> A minor suggestion, but I think you can drastically reduce the "translation" dictionary I made to something much more simple:
> 
> ```python
> DEFAULT_TRANSLATIONS = {
>     'motors': _create_motor_callable,
>     ...
> }
> ```
> 
> Where it's up to each of the callables to take in the info dictionary and return a happi JSON backend-compatible dictionary. The callable gets to choose the container, class name, device name, etc.

Perfect, i'll try that - thank you

cristinasewell commented:
So I added testing for `mpod` and `acromag` channels here, as well as the `beckoff-axis` - but, to pass the tests we'd have to first merge these two PRs and then make sure everything works fine:
https://github.com/pcdshub/pcdsdevices/pull/664
https://github.com/pcdshub/pcdsdevices/pull/690

and then we can close https://github.com/pcdshub/happi/pull/184 - because I added the info from that PR here, I think it made more sense to put them all together here as I am working with the same file and methods - but I can go back if we don't want to do so

cristinasewell commented:
I think I should get rid of the the logic handling the MPOD questionnaire entries from this PR, and do it separately when I have a better solution for https://github.com/pcdshub/pcdsdevices/pull/690 ?

ZLLentz commented:
@cristinasewell are you saying that you want to merge a PR with the various refactors you've made + the acromag stuff and then handle the MPOD section in a separate PR? I think that's a good idea if we're still discussing the MPOD stuff at pcdsdevices.

cristinasewell commented:
Yes @ZLLentz  that's what I was thinking, so we don't have to hold this PR because of MPOD - also I might have to handle the mpod questionnaire differently than I currently have anyway. 
I'm going to go ahead and remove the MPOD logic from here then, if that's ok, and if I do so this PR should be ready for a final review maybe?

ZLLentz reviewed (approved):
I think this PR is well-motivated and clean
